crypto/ct-aead/src/aes_impl.rs

```diff
@@ -23,17 +25,35 @@ where
         buffer: &mut [u8],
         tag: &Tag,
     ) -> CtDecryptResult {
-        if buffer.len() as u64 > C_MAX || associated_data.len() as u64 > A_MAX {
+        let len = buffer.len();
+
+        if len as u64 > C_MAX || associated_data.len() as u64 > A_MAX {
             return CtDecryptResult(Choice::from(0));
         }
 
         // TODO(tarcieri): interleave encryption with GHASH
         // See: <https://github.com/RustCrypto/AEADs/issues/74>
         let mut expected_tag = self.compute_tag(associated_data, buffer);
         let mut ctr = self.init_ctr(nonce);
+        let mut ciphertext = Vec::with_capacity(len);
```

Comment:
this is failing in tests, i think because, calling `Vec::with_capacity` yields a Vec which has this much allocated in its backing store, but is still logically empty, and its len is zero

Reply:
you can probably replace with `vec![0u8; len]`